### PR TITLE
Face sweep for free function API

### LINKED ISSUE #1621: Face support in free function sweep
Currently only edges/wires are supported.

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
Resolves #1621  Now it is possible to do this:
```python
from cadquery.occ_impl.shapes import *

f1 = face(rect(1, 1), circle(0.25))
f2 = face(rect(2, 1), ellipse(0.9, 0.45)).moved(x=2, z=2, ry=90)

p3 = spline(
    (f1.Center(), f2.Center()), ((0, 0, 1), (1, 0, 0))
    )

r8 = sweep((f1, f2), p3)
```
afbeelding

### CODE REVIEW

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -4910,74 +4927,164 @@ def offset(s: Shape, t: float, cap=True, tol: float = 1e-6) -> Shape:
...
+        inner_parts = []
+
+        for builder_inner in builders_inner:
+            builder_inner.Build()
+            builder_inner.Check()
+            inner_parts.append(Shape(builder_inner.Shape()))
+
+        results.append((Shape(builder.Shape()) - compound(inner_parts)).wrapped)
+
+    # otherwise construct usig wires
```
Comment (lorenzncode) [inline cadquery/occ_impl/shapes.py:5076]: typo

**Review by lorenzncode — COMMENTED** [review]
Do you think it's worth mentioning that when a sequence of shapes is provided the expected input is either a sequence of edges, wires OR faces?  Faces must not be mixed with edges or wires else the edges/wires are ignored.  Something like "Do not mix faces with other types." in loft, and sweep docstrings or a note in Free function API doc section?


Separate issue with the spline docstring -  see shapes.py L4502.  The docstring is copied from polygon.

**Review by lorenzncode — APPROVED** [review]
I've done some testing and did not find any issue.  Nice addition to the free API!  Thanks @adam-urbanczyk !

**Review by jmwright — APPROVED** [review]
Looks good. Thanks @adam-urbanczyk !

**Pull request merged**